crates/rpc/rpc-api/src/engine.rs

```diff
@@ -147,6 +148,13 @@ pub trait EngineApi {
     async fn exchange_capabilities(&self, capabilities: Vec<String>) -> RpcResult<Vec<String>>;
 }
 
+// NOTE: We can't use associated types in the `EngineApi` trait because of jsonrpsee, so we use a
```

Comment:
ah sigh...

Reply:
yeah, in theory jsonrpsee could support this, but it would be painful to implement